### ISSUE TITLE
Possible Bug - setModelSource() on ModelManager should call __destruct() to clear PHQL cache

### ISSUE THREAD

**Issue by glydetech** [phalcon/cphalcon#2235]
I was trying to change a model Source after creation, and seeing old cached data.

See full discussion here:
http://forum.phalconphp.com/discussion/1788/changing-model-source-table-dynamically

**Comment by stoiev** [comment]
+1
Phalcon 2.0.7, at least, does not provide __destruct() method

**Comment by stoiev** [comment]
A workaround to clean PHQL cache on Phalcon 2.0. Let this class on one of your include paths:

```php
<?php

class PhqlCacheCleaner extends Phalcon\Mvc\Model\Query{
    public static function clean(){
        self::$_irPhqlCache = array();
    }
}
```

And call `PhqlCacheCleaner::clean()` when you need to clean Phql cache (after setModelSource, for example).

**Comment by andresgutierrez** [comment]
Implemented in 2.1
